File: GFG/Fill up buckets/Fill up buckets.cpp

    }
};


//2nd Solution

class Solution{
public:
int totalWays(int n, vector<int>capacity) {
        long mod=1e9+7;
       sort(capacity.begin(),capacity.end());
       
       long  ans=1;
       
       for(int i=0;i<n;i++)
           ans=(ans*(capacity[i]-i))%mod;
           
        return ans;
           
       
    }
};

File: GFG/Fill up buckets/Fill up buckets.java
class Solution{
    public int totalWays(int n, int[] arr) {
        // code here
        Arrays.sort(arr);
        int mod=1000000007;
        long ans=1;
        for(int i=0;i<n;i++){
            ans=(ans*(arr[i]-i))%mod;
        }
        return (int)ans%mod;
    }
}
